src/Files.App/Views/MainPage.xaml.cs

```diff
@@ -171,6 +171,13 @@ public async void MultitaskingControl_CurrentInstanceChanged(object? sender, Cur
 			{
 				SidebarAdaptiveViewModel.PaneHolder = currentInstance;
 				SidebarAdaptiveViewModel.PaneHolder.PropertyChanged += PaneHolder_PropertyChanged;
+
+				if (currentInstance.ActivePaneOrColumn is ModernShellPage msh &&
+					msh.ItemDisplayFrame.Content is HomePage homePage &&
+					homePage.ViewModel.RequiresUpdate)
+				{
+					await homePage.ViewModel.RefreshWidgetProperties();
```

Comment:
Ideally this logic would be under `HomPage.xaml.cs`.

Reply:
Done